src/main/java/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+import bookstore.model.CartItem;
+import bookstore.model.Order;
+import bookstore.model.OrderItem;
+import bookstore.model.ShoppingCart;
+import bookstore.model.User;
+import bookstore.repository.order.OrderRepository;
+import bookstore.repository.orderitem.OrderItemRepository;
+import bookstore.repository.shoppingcart.ShoppingCartRepository;
+import bookstore.repository.user.UserRepository;
+import bookstore.service.shoppingcart.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    @Transactional
+    public OrderDto addOrder(Long userId, String address) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find user with given id: " + userId));
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart with given id: " + userId));
+        BigDecimal total = BigDecimal.ZERO;
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            BigDecimal itemPrice = cartItem.getBook().getPrice()
+                    .multiply(BigDecimal.valueOf(cartItem.getQuantity()));
+            total = total.add(itemPrice);
+        }
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.valueOf(Order.Status.PENDING.name()));
+        order.setShippingAddress(address);
+        order.setTotal(total);
+        orderRepository.save(order);
+
+        Set<OrderItem> orderItems = new HashSet<>(shoppingCart.getCartItems().size());
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            Book book = cartItem.getBook();
+
+            OrderItem orderItem = new OrderItem();
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setBook(book);
+            orderItem.setOrder(order);
+            orderItem.setPrice(book.getPrice());
+            orderItemRepository.save(orderItem);
+            orderItems.add(orderItem);
+        }
+        order.setOrderItems(orderItems);
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        shoppingCartRepository.save(shoppingCart);
+        return orderMapper.toDto(order);
```

Comment:
i think u can split this method

Reply:
agree, please divide logic in this method to smaller private methods.